articles/quickstart/native/ionic4/01-login.md

```diff
@@ -0,0 +1,443 @@
+
+To be able to make requests from your application to Auth0. Set the following origins in your [Application Settings](${manage_url}/#/applications/${account.clientId}/settings).
+
+```bash
+http://localhost, ionic://localhost, http://localhost:8100
+```
+
+the origins `http://localhost` and `ionic://localhost` are needed for Android and iOS respectively, and `http://localhost:8100` is needed you're running your application with `livereaload` option.
+
+Lastly, be sure that the **Application Type** for your application is set to **Native** in the [Application Settings](${manage_url}/#/applications/${account.clientId}/settings).
+
+## Install the Dependencies
+
+The required dependencies for using Auth0 in an Ionic application are **auth0.js**, **auth0-cordova** and **ionic-storage**. Install them with npm or yarn.
+
+```bash
+# installation with npm
+npm install auth0-js @auth0/cordova @ionic/storage --save
+
+# installation with yarn
+yarn add auth0-js @auth0/cordova @ionic/storage
+```
+
+Auth0-Cordova requires the node library `crypto`, however since Angular CLI version 6+ you are now required to use a polyfill for this library. You can accomplish this using a custom webpack configuration.
+
+To customize the webpack configuration you need to install the custom webpack builders for angular.
+
+```bash
+# installation with npm
+npm install -D @angular-builders/custom-webpack
+
+# installation with yarn
+yarn add --dev @angular-builders/custom-webpack
+```
+
+Then, create `webpack.config.js` file and add the configuration to enable the crypto pollyfill.
+
+```js
+// webpack.config.js
+
+module.exports = {
+	node: {
+		crypto: true
+	}
+};
+```
+
+In `angular.json` replace the `@angular-devkit/build-angular:dev-server` bulder with `@angular-builders/custom-webpack:dev-server`, `@angular-devkit/build-angular:browser` builder with `@angular-builders/custom-webpack:browser` and add customWebpackConfig to the build target options with the path to the webpack config file created prevoisly.
+
+```js
+// angular.json
+
+"architect": {
+  // ...
+  "build": {
+    "builder": "@angular-builders/custom-webpack:browser",
+    "options": {
+      "customWebpackConfig": {
+        "path": "webpack.config.js"
+      },
+      // ...
+    }
+  },
+  // ...
+  "serve": {
+    "builder": "@angular-builders/custom-webpack:dev-server",
+    // ...
+  }
+  // ...
+}
+```
+
+### Add Cordova Plugins
+
+You must install the `SafariViewController` plugin from Cordova to be able to use universal login. The downloadable sample project already has this plugin added, but if you are embedding Auth0 in your own application, install the plugin via the command line.
+
+```bash
+# installation with cordova
+ionic cordova plugin add cordova-plugin-safariviewcontroller
+
+# installation with npm
+npm install --save @ionic-native/safari-view-controller
+
+# installation with yarn
+yarn add @ionic-native/safari-view-controller
```

Comment:
when I used yarn, it didnt actually add the plugin under cordova/plugins in my package.json

Reply:
For me this is added when I run `ionic cordova plugin add cordova-plugin-safariviewcontroller`.

Reply:
That might be the case, but there are 3 options to install it here.  The yarn route didn't work, at least not for me, and when viewing their official docs, they don't seem to list it as an option either https://ionicframework.com/docs/native/safari-view-controller